cpp/include/raft/spatial/knn/faiss_mr.hpp

```diff
@@ -0,0 +1,333 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
```

Comment:
Thanks for opening the issue w/ the FAISS project: https://github.com/facebookresearch/faiss/issues/2097. 

We also need to reference that issue here with a comment that explains why we've temporarily copied this code and how we can remove the copied code once that issue is resolved.